src/MessageReader.test.js

```diff
@@ -45,6 +45,24 @@ describe("MessageReader", () => {
       });
     });
 
+    it("parses JSON", () => {
+      const reader = new MessageReader("#pragma rosbag_parse_json\nstring dummy");
+      const buff = getStringBuffer('{"foo":123,"bar":{"nestedFoo":456}}');
+      expect(reader.readMessage(buff)).toEqual({
+        dummy: { foo: 123, bar: { nestedFoo: 456 } },
+      });
+
+      const readerWithSpaces = new MessageReader(" #pragma rosbag_parse_json  \n  string dummy");
+      expect(readerWithSpaces.readMessage(buff)).toEqual({
+        dummy: { foo: 123, bar: { nestedFoo: 456 } },
+      });
+
+      const readerWithNewlines = new MessageReader("#pragma rosbag_parse_json\n\n\nstring dummy");
+      expect(readerWithNewlines.readMessage(buff)).toEqual({
+        dummy: { foo: 123, bar: { nestedFoo: 456 } },
+      });
+    });
```

Comment:
Maybe also add a test for multiple definitions, like here: https://github.com/cruise-automation/rosbag.js/blob/c8d16e888c297b87956ce4097166862bad6a504f/src/MessageReader.test.js#L246-L252 E.g. in particular, if you have a pragma at the end of a definition file at the very last line, then it shouldn't affect the next definition (we should instead just ignore it).